Sources/System/FileOperations.swift

```diff
@@ -158,14 +158,23 @@ extension FileDescriptor {
     into buffer: UnsafeMutableRawBufferPointer,
     retryOnInterrupt: Bool = true
   ) throws -> Int {
-    try _read(into: buffer, retryOnInterrupt: retryOnInterrupt).get()
+    try _readNoThrow(into: buffer, retryOnInterrupt: retryOnInterrupt).get()
   }
 
+  // NOTE: This function (mistakenly marked as throws) is vestigial but remains to preserve ABI.
   @usableFromInline
   internal func _read(
     into buffer: UnsafeMutableRawBufferPointer,
     retryOnInterrupt: Bool
   ) throws -> Result<Int, Errno> {
+    _readNoThrow(into: buffer, retryOnInterrupt: retryOnInterrupt)
+  }
+
+  @usableFromInline
```

Comment:
Unfortunately yes, and on ABI stable platforms, `read` will need to dispatch to either `_read` or `_readNoThrow`, depending on the platform & version we're running on. 

Given the importance of this particular entry point, I think it's worth considering marking the new non-throwing variant `@_alwaysEmitIntoClient @inline(never)` to prevent the availability check.

Otherwise we'll need to mess with `#if SWIFT_PACKAGE` to enable/disable availability checking based on whether or not we're building the ABI stable variant.

I think it's okay to land this without resolving the availability mess -- ABI stable builds aren't possible to do within this repo, and I wouldn't want to force Simon to have to deal with a tricky problem based on partial info and no tooling. It's tricky enough to get right when we have the benefit of the ABI checker and a working build system. (Unless he really really wants to do it, of course!)

Reply:
> I wouldn't want to force Simon to have to deal with a tricky problem based on partial info and no tooling. It's tricky enough to get right when we have the benefit of the ABI checker and a working build system. (Unless he really really wants to do it, of course!)

😅 sounds like I might be opening a can of worms here. It sounds like merging this PR without addressing the ABI issue might be the quickest solution. After merging though, I'm happy to collaborate on how to please the ABI checker, but I confess this isn't something I've done before.